src/components/Nav/SiteMap.jsx

```diff
@@ -0,0 +1,136 @@
+
+const DIVIDE_CATEGORIES = 6;
+
+function SiteMap(props) {
+  useEffect(() => {
+    fetch('http://localhost:3000/data/navCategories.json')
+      .then(response => response.json())
+      .then(result => {
+        const addedEmptyCategories = 2 * DIVIDE_CATEGORIES - result.length;
+        if (
+          result.length > DIVIDE_CATEGORIES &&
+          result.length < 2 * DIVIDE_CATEGORIES
+        ) {
+          for (let i = 0; i < addedEmptyCategories; i++) {
+            result.push({});
+          }
+        }
+        setCategories(result);
+      });
+  }, []);
+  const [categories, setCategories] = useState([]);
+
+  const TopWrapper = styled.div`
+    position: absolute;
+    top: 91px;
+    width: 100%;
+    display: ${props.display ? 'flex' : 'none'};
+    justify-content: center;
+    background-color: #333;
+    padding-top: 50px;
+    padding-bottom: 50px;
+  `;
+
+  const Wrapper = styled.div`
+    width: 1100px;
+    display: flex;
+    flex-direction: column;
+    align-items: start;
+  `;
+
+  const MenuTitle = styled.p`
+    font-size: 1.5em;
+    color: #fff;
+    line-height: 1.1;
+    padding: 0;
+    margin: 0;
+  `;
+
+  const SiteMapUpperWrapper = styled.div`
+    width: inherit;
+    display: flex;
+    justify-content: space-between;
+    margin-bottom: 50px;
+  `;
+
+  const SiteMapLowerWrapper = styled.div`
+    width: inherit;
+    display: flex;
+    justify-content: space-between;
+    margin-top: -230px;
+  `;
+
+  const SubCategories = styled.ul`
+    width: 170px;
+    padding: 0;
+    text-align: center;
+  `;
+
+  const SubCategory = styled.li`
+    padding-bottom: 10px;
+  `;
+
+  const SubCategoryLink = styled(Link)`
+    color: #bbb;
+  `;
+
+  const SubCategoriesTitle = styled.p`
+    text-align: center;
+    border-width: 1px 0;
+    border-color: rgba(68, 68, 68, 0.5);
+    border-style: solid;
+    padding: 12px 0;
+    font-size: 1.2em;
+    color: #fff;
+    margin: 20px 0;
+  `;
+
+  return (
+    <TopWrapper>
+      <Wrapper>
+        <MenuTitle>SITEMAP</MenuTitle>
+        <SiteMapUpperWrapper>
+          {categories.slice(0, DIVIDE_CATEGORIES).map(category => (
+            <SubCategories key={category.id}>
+              <SubCategoriesTitle>{category.categoryName}</SubCategoriesTitle>
+
+              {category.subcategories.map(subcategory => (
+                <SubCategory>
+                  <SubCategoryLink>{subcategory.categoryName}</SubCategoryLink>
+                </SubCategory>
+              ))}
+            </SubCategories>
+          ))}
+        </SiteMapUpperWrapper>
+
+        <SiteMapLowerWrapper>
+          {categories
+            .slice(DIVIDE_CATEGORIES, categories.length)
+            .map(category => (
+              <SubCategories key={category.id ? category.id : null}>
+                {category.categoryName ? (
+                  <SubCategoriesTitle>
+                    {category.categoryName}
+                  </SubCategoriesTitle>
+                ) : null}
```

Comment:
```suggestion
              <SubCategories key={category.id}>
                {category.categoryName && (
                  <SubCategoriesTitle>
                    {category.categoryName}
                  </SubCategoriesTitle>
                )}
```
- key에서 null처리 해주면 안될거 같습니다 혹시 의도가 무엇인가요?
- 그리고 조건부 렌더링시에 true 값만 렌더링 할거라면 &&을 쓰시면 됩니다

Reply:
밑에 lowerwrapper랑 upperwrapper랑 들어가는 ul 개수가 같아야
실제 페이지랑 똑같이 보여서요!

현재 flex가 space-between으로 되어있는데
밑에 개수가 위에 개수보다 작을 때
그냥 빈 ul를 넣으려고 합니다.